### PR TITLE
Fix coverage aggregation script

### PR DESCRIPTION
**Pull request by andrewkroh**
Fix a possible bug with coverage aggregation when the output file is
globbed and becomes an input resulting in a loop that fills the disk.

### CODE REVIEW

**Comment by andrewkroh** [comment]
Fixes elastic/infra#640. I couldn't reproduce the issue locally so I [built this branch](http://build-eu-00.elastic.co/view/Beats/job/beats/237/) on Jenkins and it didn't explode.

**Comment by andrewkroh** [comment]
Once this merges, re-enable the [job](http://build-eu-00.elastic.co/view/Beats/job/beats/) and :pray: .

**Comment by andrewkroh** [comment]
Looks like my solution breaks Windows. Could one of you Python experts implement a cross-platform friendly solution. Otherwise I'll pick it up tomorrow.

**Comment by andrewkroh** [comment]
I updated the script to compare absolute file paths and it seems to be working.

File: libbeat/scripts/aggregate_coverage.py [hunk 1]
```diff
@@ -28,9 +28,10 @@ def main(arguments):
     args.outfile.write('mode: atomic\n')
     for m in matches:
         with open(m) as f:
-            for line in f:
-                if not line.startswith('mode:'):
-                    args.outfile.write(line)
+            if os.path.abspath(args.outfile.name) != os.path.abspath(f.name):
```
Comment (urso) [inline libbeat/scripts/aggregate_coverage.py]: why not compare before opening the file? os.path.abspath(args.outfile.name) != os.path.abspath(m)

Comment (andrewkroh) [inline libbeat/scripts/aggregate_coverage.py]: That would make more sense. Will change.

Comment (andrewkroh) [inline libbeat/scripts/aggregate_coverage.py]: I made that change.

**Pull request merged**